openvdb_ax/openvdb_ax/codegen/StandardFunctions.cc

```diff
@@ -0,0 +1,1982 @@
+            // The generator takes an unsigned int, which is not guaranteed to be the
+            // same size as size_t.
+            //
+            // So, we must convert it. I should note that the OpenVDB math libraries will
+            // do this for us, but its implementation static_casts `size_t` to `unsigned int`,
+            // and because `boost::hash` returns a binary copy of the original
+            // double-precision number in almost all cases, this ends up producing noticable
+            // patterns in the result (e.g. by truncating the upper 4 bytes, values of 1.0,
+            // 2.0, 3.0, and 4.0 all return the same hash value because their lower 4 bytes
+            // are all zero).
+            //
+            // We use the `hashToSeed` function to reduce our `size_t` to an `unsigned int`,
+            // whilst taking all bits in the `size_t` into account.
+            const uint32_t uintseed = hashToSeed<uint32_t>(hash);
+            return Rand::rand(&uintseed);
+        }
+
+        static double rand(int32_t seed)
+        {
+            const uint32_t uintseed = static_cast<uint32_t>(seed);
+            return Rand::rand(&uintseed);
+        }
+    };
+
+    return FunctionBuilder("rand")
+        .addSignature<double()>((double(*)())(Rand::rand))
+        .addSignature<double(double)>((double(*)(double))(Rand::rand))
+        .addSignature<double(int32_t)>((double(*)(int32_t))(Rand::rand))
+        .setArgumentNames({"seed"})
+        // We can't constant fold rand even if it's been called with a constant as
+        // it will leave the generator in a different state in comparison to other
+        // threads and, as it relies on an internal state, doesn't respect branching
+        // etc. We also can't use a different generate for constant calls as subsequent
+        // calls to rand() without an argument won't know to advance the generator.
+        .setConstantFold(false)
+        .setPreferredImpl(op.mPrioritiseIR ? FunctionBuilder::IR : FunctionBuilder::C)
+        .setDocumentation("Creates a random number based on the provided "
+            "seed. The number will be in the range of 0 to 1. The same number is "
+            "produced for the same seed. Note that if rand is called without a seed "
+            "the previous state of the random number generator is advanced for the "
+            "currently processing element. This state is determined by the last call to "
+            "rand() with a given seed. If rand is not called with a seed, the generator "
+            "advances continuously across different elements which can produce non-"
+            "deterministic results. It is important that rand is always called with a "
+            "seed at least once for deterministic results.")
+        .get();
+}
+
+inline FunctionGroup::Ptr axsignbit(const FunctionOptions& op)
+{
+    return FunctionBuilder("signbit")
```

Comment:
I'd go with
```sign```
which would include integers and return 1/0/-1.

Reply:
Added sign which implements signum. Left signbit purely for the +-0.0 case